bitcoin/src/util/mod.rs

```diff
@@ -96,6 +48,9 @@ pub mod address {
 #[deprecated(since = "0.30.0", note = "Please use crate::bip32")]
 pub use crate::bip32;
 
+#[deprecated(since = "0.30.0", note = "Please use crate::Error")]
+pub use crate::error::Error;
```

Comment:
Does this work? AFAIR, we need to define another type. I am also okay with multiple breaking changes in this release. We should add a big note guiding users to which types map to which new types in the release notes. I am certain that with the compiler and the release notes, users should be able to debug things.

Reply:
To be honest, even though I've previously hacked up demo code to work this out, this morning I could not remember what works and what does not in regards to deprecated. I believe we have stuff on master that does not work (totally my fault). I'm in the process of doing so many "moving things" PRs that I have put off fixing up the deprecation stuff. I will definitely fix it before 0.30 release though. Makes review a bit hard I know, sorry about that.

Reply:
Confirming that this does not work. Only deprecation of functions works, as far as I can work out. I've remove the deprecation from this PR and will do a PR to remove all the other misleading ones I added recently to `util`. Thanks for picking this up @sanket1729

Reply:
I believe `#[deprecated] type Foo = somewhere::Foo;` should work

Reply:
I've gone with just not re-exporting. There is going to be so much module movement in this next release that re-exporting some things an not others seems like a waste of effort, users are likely going to have to look at the code and release notes to work this out - its not going to be super friendly but I don't think its avoidable.